### PR TITLE
[FIX] website: make dynamic snippet compatible with visible on mobile

### PR DESCRIPTION
**Pull request by bso-odoo**
Both dynamic snippets and visible on mobile use the `d-none` class:
- dynamic snippets activate it by default and only remove it when the
content is ready to be displayed
- visible on mobile makes the section `d-none` when it must be hidden on
mobile and the class is contradicted by another class which is only
taken into account non-mobile

This commit introduces a dedicated class to temporarily hide the dynamic
snippets.

task-2930503

### CODE REVIEW

File: addons/website/static/src/snippets/s_dynamic_snippet/000.scss [hunk 1]
```diff
@@ -1,4 +1,7 @@
 .s_dynamic {
+    &.o_dynamic_empty {
+        display: none;
+    }
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_dynamic_snippet/000.scss:4]: Would not this be ignored if the "hide in mobile" option is enabled as d-md-block uses `!important` ?

File: addons/website/views/snippets/s_dynamic_snippet.xml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
     <template id="website.s_dynamic_snippet_template">
+        <!-- TODO In master, replace `d-none` by `o_dynamic_empty` -->
```
Comment (qsm-odoo) [inline addons/website/views/snippets/s_dynamic_snippet.xml:4]: Indeed, to be done in master. But in stable, you can already patch the `onBuilt` of the related option to replace d-none by the right class once the snippet is added in the DOM (best not wait for the public widget to do it (even though still necessary to do it there for compatibility), `onBuilt` should normally be safer regarding editor history etc).

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/snippets/s_dynamic_snippet/000.scss (hunk 1 above)
Comment (bso-odoo) [reply addons/website/static/src/snippets/s_dynamic_snippet/000.scss]: :facepalm: I actually thought about this, but then I could not spot the difference.
Re-testing now at a cooler temperature I actually saw it. Sorry...

**Comment by bso-odoo** [comment]
@qsm-odoo I just noticed I forgot to ping you after making that change.

**Review by qsm-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**